### DISCUSSION TITLE
Chati chat

### DISCUSSION THREAD

**Discussion by likeproblem** [likeproblem/RestockRecolour#3] [General]
@UncleMateo here's general chat
So I created page for wiki with part association, edit it with your own, and also check out categories I made inside those two files
And wiki page for tools I use, category text creator included

**Comment by likeproblem** [comment]
https://github.com/likeproblem/RestockRecolour/wiki/Part-association-list#unclemateo
Just edit the wiki

**Comment by likeproblem** [comment]
Im away from pc right now, so I can't work for another several hours

**Comment by UncleMateo** [comment]
![image](https://github.com/user-attachments/assets/c76db658-f684-4ba5-9b8e-3645d755282c)
![image](https://github.com/user-attachments/assets/0d57d74c-51b1-4407-8a79-83ce991f27dc)
Making progress... these probe cores are basically done but there's an issue with the command chair, I'm going to go over the config again and see if I can't fix it.. basically, the TU slider retains the metallic and paint options from the Stock Turd for some reason.. not sure why.  Those options don't do anything, but once you switch to the working BobRoss, clicking the slider doesn't do anything.  Like I said, I'll investigate. 

Should upload those 5 working parts soon.

**Reply by UncleMateo** [reply]
Also, I don't know about for you, but I haven't been able to get on the forums pretty much all day.. a bit worrisome.

**Reply by likeproblem** [reply]
weird, can you screenshot the config for chair? also ngl I never paint insides nor "connectors with black dots"

**Reply by UncleMateo** [reply]
![image](https://github.com/user-attachments/assets/04cc68e8-d4dc-4fc1-8ad5-1f3a73d529e1)
As you can see, I did as you said to try and correct against that glitch, but it didn't completely work.
Do we need a separate config file for the false-TU-recolors?  Or will they load correctly if placed in the main config?

Also, I'll try to tone down my inner Bob Ross and stick more to the guidelines.  The two ReStock+ cores I'm pretty pleased with, they have their circuits an guts unpainted, but the three from ReStock proper I think could use more attention

**Reply by likeproblem** [reply]
You forgot the line which removes modules before hand, check dart for example, it has that line in Default in start, and the line should be in Guidelines

**Reply by likeproblem** [reply]
Add this ```!MODULE[KSPTextureSwitch],* {}``` to Default BEFORE MODULE{...}

**Reply by likeproblem** [reply]
Like this:
![Screenshot_20241010-173754](https://github.com/user-attachments/assets/02b5caa7-e393-4288-9f1d-cb028d734d2c)

**Reply by UncleMateo** [reply]
Testing now, also tweaked some of the probe cores to have less paintable guts.  If this works, my friend, you solved my KerbalFoundries conundrum that's been haunting me for.. I don't know, two months?  Anyways, thank you!

**Reply by UncleMateo** [reply]
...it was basically the same issue, or close enough I think this might possibly fix it.

**Reply by likeproblem** [reply]
So turd was basically coloring those parts falsely?

**Reply by UncleMateo** [reply]
Hey it worked perfectly! 

As to your question: Well, not exactly.. basically, KerbalFoundries at some point was updated and integrated with TexturesUnlimited.. so it already has all these texture sets, which resulted in a second TU slider that broke the recoloring in the same way as this issue was doing.. if I understand my MM syntax, I think what "!MODULE[KSPTextureSwitch],* {}" does is deletes any other TextureSwitch modules?  Well, if so, that's the solution I was trying to find.. see, with that mod there's only like 30 parts, but of those, 22 or so have this issue.

**Reply by likeproblem** [reply]
Yeah, that line removes all existing switches, hence ,*, and we also need to be super late using zzzzz_, so we made sure TURD doesn't override us again

**Comment by likeproblem** [comment]
@UncleMateo also I noticed some "not so falsely" colored parts, I think only jet engines and intakes, lack "propellers", for example Juno is transparent on bottom

So guess what, we need to document them too and fix

**Reply by UncleMateo** [reply]
There is some kind of visual gitch with the depthmasks that ReStock adds to some of the aviation parts where some portion of the part might look wrong or transparent in the VAB/SPH but once loaded ingame it looks fine.  That being said, I'm also worried I may have disregarded parts with broken or false recolors at the start of my count so I will do another sweep regardless.

This morning I'm trying to get the hang of the GitDesktop, and Blender as well.  Honestly I've never really used Blender so far, I generally use a trial and error recoloring process combined with trying to interpret the exportedUVmaps.  But now I've got the bottoms of every single one of those probe cores recolored in a way I don't want and would rather fix it faster than my tedious and inefficient way, so I'll use the whole vertically-flipped-jpeg process through Blender to see what I need to do.  Once I finish all that I should be able to update the repository and the wiki and then, before I start on the next wave of parts, I'll look through **all** the parts again and try and track down anything glitchy, anything that I missed.  I will make sure to look carefully over the aviation type parts that TURD recolors and ReStock doesn't really mess with (except like I said, to add depth masks to appropriate parts) to make sure they're all actually working.

**Reply by likeproblem** [reply]
Sure! I very recommend github desktop, very easy to use, just clone the rep, and then just edit files and it will show up in app, then commit and push origin, quite easy. Sometimes you will also need to pull, since I also push, but don't worry, pull works the way it will mostly not change your changes, but still it is better to backup files
Blender is very useful! I can save so much time with it, before it I needed to color using svgs from exportedUVMaps, and trial and error at least 4 game launches(20 minutes per launch). But now I can minimise it to barely 2, and second one is mostly changing brightness variables

**Comment by UncleMateo** [comment]
Alright so I managed to figure out GitDesktop.. it is indeed quite easy to use.  Really my fear was that I might overwrite/delete something in my bumbling newness to Git, but it looks like it's (moderately) idiot proofed.  Quite easy to use.  Managed to clean up those parts a bit, but not through Blender, sadly I've had a hard time installing it, and of course every time I was trying to follow a link regarding that, the forums were having issues... they seem alright now, though, so I'm trying to get it installed correctly.  So I wasted some time trying and failing to install the MU reader module, but I'm back on track.  Will update the wiki shortly with pictures, at the same time I'm going to recompile the list.  I realized when I first made the MasterList, I may have changed the order of the parts from that of the 'Name' ordinal, simply because I moved some that were done to the top of the list without considering the consequences.  So I might have to fix that (if it matters).

**Reply by likeproblem** [reply]
Hey, just in case I recommend to save recolours with .pdn from paint.net, so it is easy to modify

**Reply by likeproblem** [reply]
Hey, won't mind if I convert your dds to png+pdn?

**Reply by UncleMateo** [reply]
Sorry for taking so long to reply, I ended up being really busy the last ~36 hours and wasn't able to get much work in.  If you want to convert to another format, that's totally fine by me, I just typically use GIMP as much as I can, and mostly just use DDS when possible, but I can convert back an forth just fine, if it works better for you.  Ultimately though, we should switch them all back to DDS when the project is closer to completion so the loading burden on KSP isn't as bad, since DDS is a lot more compact, and there's a lot of parts (and therefore textures).

**Reply by likeproblem** [reply]
Yea sure, I just use paint.net, and therefore also save .pdn files for easier modification in future(it saves layers etc)

**Reply by UncleMateo** [reply]
I'm curious, do you have to vertically flip .pdn files as you do .pngs?

**Reply by likeproblem** [reply]
Of course, pdn is just png but with layers, so you can flip it, but if you don't you need to still flip output png file
Pdn is nothing more than project file

**Comment by Spartwo** [comment]
Bit of a development aid I made for BD TURD. Adjust the patch file and it'll show a list of transform names in the ksp right click window. Saves a bunch of importing to blender. [TransformLister.zip](https://github.com/user-attachments/files/17365266/TransformLister.zip)

**Reply by likeproblem** [reply]
Transform names as different meshes included into model?

**Reply by Spartwo** [reply]
Meshes and also stuff like engine fx transforms but restock has decent naming conventions so you'll be able to tell which is which.

**Reply by likeproblem** [reply]
Sounds great! Though it might be obsolete since exportedUVMaps also gives you mesh names, but thanks anyway!

**Comment by likeproblem** [comment]
@UncleMateo since you added @Spartwo as contributor to part assocciation list, do you want me to add him as contributor too to the rep?

**Reply by UncleMateo** [reply]
I was thinking along those lines, but it all depends how you want to structure this.  Really what's important is our ability to coordinate what parts we are each working on to avoid having two people doing the same part at the same time.

Also..

I don't know what I've done, but 4 of my batteries broke when I moved their configs over.. I've been looking for the past hour but can't figure out what I did.  Trying to fix it but can't seem to figure out where the error lies.  Had a backup, but it looks like it's the same as the current version.  Kinda stumped but not giving up.  I just need to step away for a minute so I don't gorilla-slam my computer.

**Reply by likeproblem** [reply]
same here man

I think about creating cfgs with the creator, it usually works

also @Spartwo is invited to collaboration now

**Reply by likeproblem** [reply]
Found an issue for batteries, the } is misplaced

**Reply by likeproblem** [reply]
oh my god, there's so much cfg errors

I might need to reserve to config maker

**Comment by likeproblem** [comment]
@UncleMateo nailed it. Batteries fixed by just using config maker plus copying some values other. Will commit after I figure out problem with Spartwo's parts

**Reply by likeproblem** [reply]
Pushed another commit fixing most @Spartwo parts, only science bay is weirding out, after I fix that I should be able to announce 61part on thread :D

**Reply by Spartwo** [reply]
All TURD mods, regardless of if they are stock recolour, go into the "TURD/" folder. It's standard practice for all the other mod compatibility packs

**Reply by Spartwo** [reply]
Rapier though yea, the engine shroud is very dark because I never made a base coat for it and just turned down the detail

**Reply by Spartwo** [reply]
> Rapier though yea, the engine shroud is very dark because I never made a base coat for it and just turned down the detail

If its not already done I can come back to this in a few and make that change. Probably knock out a few other science parts or whatever is still on the go by that time.

**Comment by Spartwo** [comment]
"Science Jr looking weird"
![Image](https://i.imgur.com/ZmU6SVN.jpg)

It should be coming out like this, dark(basecoat again) but any unpainted wires or internals are intended

**Reply by likeproblem** [reply]
I meant it's fully dark red for me, which means texture is not loaded properly

**Reply by likeproblem** [reply]
Can I also ask a question? Why rapier has this new -base.dds file for? Fully white variant?

**Reply by Spartwo** [reply]
basecoats make the painting more consistent. Brightness of the underlying texture leaks through painted parts which throws off saved patterns.

I am now in a situation where I can check and I saved those in the wrong dds type. My next push addresses that but it looks fine. Taking screenshots for the gallery.

**Reply by likeproblem** [reply]
I am not sure to which issue you are referring, but thanks anyway

About the gallery... It's screwed, I will need to remade it with newest pics plus compact multiple into one

**Reply by Spartwo** [reply]
Oh...
[Screenshots.zip](https://github.com/user-attachments/files/17380450/Screenshots.zip)

**Reply by likeproblem** [reply]
Looks great! Also I would recommend if it's possible to uncolor inside bell of engine for consistency and because generally paint would just melt instantly 

I will have to figure put science bay and then it is ready to announce

**Reply by Spartwo** [reply]
Why even paint the engine bell if you're not going to paint the visible section

**Reply by likeproblem** [reply]
I guess makes sense, but I wouldn't recommend doing it for regular engines because they usually get ovveridden by engineCore anyway

**Comment by Spartwo** [comment]
Question: Should I be writing in restock+ configs with the rest of them. Restock+ trusses share masks with regular is the reason

**Reply by likeproblem** [reply]
Yes, because MM's patches if don't find part for modification, doesn't even throw error as I know of

**Reply by likeproblem** [reply]
Also edit part association list with your parts you are working on, to avoid confusion

**Comment by likeproblem** [comment]
@Spartwo how do you work this fast?? I just wanted to make sure all counters are correct and you flashbang me with this amount of parts

**Reply by Spartwo** [reply]
I've gotten pretty fast from doing BD TURD

**Comment by likeproblem** [comment]
@UncleMateo @Spartwo did ksp forum crash for you? 
![image](https://github.com/user-attachments/assets/71f391cf-d590-453b-9ff2-4016ef3ed784)

**Reply by likeproblem** [reply]
![image](https://github.com/user-attachments/assets/3104f93a-912a-4182-9c9d-d63fd00ea87b)
another one

**Reply by likeproblem** [reply]
I can't reach it about a day now, so ~7 hours

**Reply by UncleMateo** [reply]
KSP forums have been going up and down constantly for me.. if it's down I check it again in 10 minutes and sometimes its fine.  I worry that they're fixing on kraken'ing themselves to smithereens.

**Reply by UncleMateo** [reply]
Well now whenever its up I get some kind of undefined configuration error.. can't seem to get any pages to load.

**Reply by likeproblem** [reply]
still down

**Reply by UncleMateo** [reply]
I worry that the forums could be dead for good.  I hope they aren't.. but it don't look good.

**Reply by likeproblem** [reply]
I never encountered such error, but I know some fellows on reddit where it works for them
 
If it's actually dead, I heard admins have a back up forum, but without all the data

**Reply by UncleMateo** [reply]
I feel like something happened before where the forums got more or less purged.  In fact I think it's happened twice so far, once in the earlier years, and once around 2016 or so where not everything was lost.  So there is hope.

If the forums did go down and were not replaced, or were not replaced quickly, though... that might slowly kill the modding community, even if CKAN and SpceDock are left unscathed.  Discord has just atomized so much of what used to be on forums, I worry what the result would be without a centralized discussion space.

Well, either way, I'm gonna keep on truckin', I've only just begun making Restock Recolours!

**Reply by likeproblem** [reply]
Yeah, we need to finish this, we have GitHub at least, but I just wish I had more free time and could make them faster

**Reply by likeproblem** [reply]
48 hours now

**Reply by likeproblem** [reply]
72 hours

**Reply by UncleMateo** [reply]
I worry the forums aren't coming back.. but also, can Take2 really claim they're still working on KSP2 if there's nowhere for KSP2 players to go to get support?  Without the forums, can they _legally_ claim that they are still working on KSP2, when they're having a garage sale and liquidating all associated assets? Anyways, just a thought.

**Comment by likeproblem** [comment]
@Spartwo 
![screenshot50](https://github.com/user-attachments/assets/38742f15-2219-4ae2-a0b1-9bb36f7f06c7)
for some reason radial science jr is ***TRANSPARENT***

**Reply by Spartwo** [reply]
It's the same masking plane bug as the jet engines and big 1x6 solars get in normal TURD. It clears on load.

**Reply by likeproblem** [reply]
nvm sorry it is nonexistant in flight

**Reply by UncleMateo** [reply]
I think I can explain.. this is sort of a glitch within KSP's code that doesn't allow for any mod adding depthmask effects, or effects that depend on them, to be rendered in the SPH/VAB.  There's A LOT of Aero parts that get some kind of depth mask related patches especially.  But I think even some moving parts depend on them.

**Comment by likeproblem** [comment]
Dammit, I would have announced 77 part already if not the forum dying

**Reply by UncleMateo** [reply]
Just broke past 80- per my calculations we just hit 25% of all parts

**Reply by likeproblem** [reply]
That's great! I will try to get parts done, never enough time to do anything serious:(

**Comment by likeproblem** [comment]
@Spartwo can I ask what those cockpit but with recolored window variations mean?

**Reply by Spartwo** [reply]
![image](https://github.com/user-attachments/assets/1b95f718-c4bb-459a-a76b-a13f1a314e04)

**Reply by UncleMateo** [reply]
Do the emissive lights still work if you do that?  I was under the impression it would screw them up, but now I don't know what to think!

**Reply by likeproblem** [reply]
If you somehow edit -e file, maybe, because as I know of emissive file overlaps main texture

**Reply by likeproblem** [reply]
(but I am not an expert)

**Reply by likeproblem** [reply]
Just in case, if you actually make it, implement it as another texture set(2 swappable texture sets in total)

Im the type of guy who always keep lights on, so I might never need it lol, and original TURD I think didn't do it

**Reply by Spartwo** [reply]
That's the plan! Secondary texture is what I did in that screenshot too

Standard emissive maps take over, the lights still light if that's your concern

**Reply by Spartwo** [reply]
![image](https://github.com/user-attachments/assets/a89fb0d9-65d5-486b-b74e-37f4fded29c5)
![image](https://github.com/user-attachments/assets/7123e9c0-89de-4b76-b9fd-55c114586028)

**Reply by likeproblem** [reply]
I guess it is only beauty until you turn on lights 😆

**Reply by UncleMateo** [reply]
My mind is blown.. I didn't know you could even do that

**Comment by likeproblem** [comment]
I sincerely apologize, but I barely get any free time lately, so I might be very slow with contributions, all free time last days I litteraly spent counting parts. :[

**Reply by UncleMateo** [reply]
No need to apologize, this is a tremendous project.  I feel like I took a lot time to get rolling with my contributions, and now that forums seem to be dead, I feel guilty that I took so long to even get started on Restock, and that I didn't get more done with my Felota project before the inevitable end.  I mean, I know there's rumors they could rebuild the forums, sans content.. but my experience has taught me to always plan for the worst.

I apologize in advance, because I know I'm going to have to shift some of the time I've been spending on the RestockRecolour back to the Felota Project given the circumstances: I think if the forums are down for a solid week I'm going to start polishing my completed recolors and putting them up on spacedock/CKAN, one-by-one.  There's no real other way to get people to have easy access to mods now that the forums are in a state of flux, I mean github isn't the most searchable site, especially for the layman.  

Well, also should note I made a more comprehensive master list, basically I included any Stock/BG/MH parts that aren't in restock as well, making note of the recolors that work, and those that don't.  Since the wiki is going to be more for public consumption, I thought updating the list with the actual in-game names of each part, and putting them in in-game 'name' order, would make things simpler.  But that's something for another day.

**Reply by likeproblem** [reply]
Ooh, new wiki list sounds great great! Thanks for spending your time documenting them, that is definitely a pain to go through each single one of them

**Comment by UncleMateo** [comment]
Hey I just want to say, I finally figured out your config-maker, @likeproblem, and it is truly awesome, you've made the business of making these configs a lot more streamlined
I love it.

**Reply by likeproblem** [reply]
Glad it helped! You can also .gitignore it and customize it to your liking!(Or put otside github folder and mess with paths.txt)

**Comment by likeproblem** [comment]
Also, I think of creating another python script, it will easily make releases, just copy paste TU_Restock, but remove folders, files and extensions, such as .pdn, obviously configurable, and maybe even zip it. What do you think?

**Comment by likeproblem** [comment]
@UncleMateo the wiki looks sick!

**Reply by likeproblem** [reply]
Also a note, please put spaces between lines or [] and ### lines wont work

**Comment by likeproblem** [comment]
@Spartwo @UncleMateo retiring this chat for *hopefully* more organised per-topic disscusions
